specs/core/0_beacon-chain.md

```diff
@@ -809,12 +838,16 @@ For each one of these attestations:
 
 Extend the list of `AttestationRecord` objects in the `state` with those included in the block, ordering the new additions in the same order as they came in the block.
 
-Let `curblock_proposer_index` be the validator index of the `block.slot % len(get_shards_and_committees_for_slot(state, block.slot)[0].committee)`'th attester in `get_shards_and_committees_for_slot(state, block.slot)[0]`, and `parent_proposer_index` be the validator index of the parent block, calculated similarly. Verify that an attestation from the `parent_proposer_index`'th validator is part of the first (ie. item 0 in the array) `AttestationRecord` object; this attester can be considered to be the proposer of the parent block. In general, when a beacon block is produced, it is broadcasted at the network layer along with the attestation from its proposer.
+### Verify proposer signature
 
-Additionally, verify and update the RANDAO reveal. This is done as follows:
+Let `proposal_hash = hash(ProposalSignedData(fork_version, block.slot, 2**64 - 1, block_hash_without_sig))` where `block_hash_without_sig` is the hash of the block except setting `proposer_signature` to `[0, 0]`. 
+
+Verify that `BLSVerify(pubkey=get_proposer(state, block).pubkey, data=proposal_hash, sig=block.proposer_signature)` passes.
```

Comment:
```suggestion
Verify that `BLSVerify(pubkey=get_proposer(state, block.slot).pubkey, data=proposal_hash, sig=block.proposer_signature)` passes.
```

* pass in `block.slot` to match the updated `get_beacon_proposer` params

Reply:
Fixed.